_episodes/l1-04-managing_contributions.md

```diff
@@ -0,0 +1,166 @@
+
+objectives:
+- Create a pull request from a branch within a repository.
+- Create a pull request from a forked repository.
+keypoints:
+- Forks and pull requests are GitHub concepts, not git.
+- Pull request can be opened to branches on your own repository or any other fork.
+- Some branches are restricted, meaning that PR cannot be open against them.
+- Merging a PR does not delete the original branch, just modifies the target one.
+- PR are often created to solve specific issues.
+---
+
+## Pull Requests
+Pull requests are a GitHub feature which allows collaborators tell each other about changes that have been pushed to a branch in a repository. Similar to **issues**, an open pull request can contain discussions about the requested changes and allows collaborators to review proposed ammendments and follow-up commits before changes are either rejected or accepted and merged into the base branch.
+
+> The term "Pull Request" may sound counterintuitive because, from your perspective, you're not actually requesting to pull anything. Essentially it means “Hey, I have some changes I would like to contribute to your repo. Please, have a look at them and pull them into your own.”
+>
+> You may see the term `merge request` instead of `pull request`. These are essentially the same thing. Different platforms use different terms but they're both asking the receiver of the request to review those changes prior to merging them.
+{: .callout}
+
+There are two main workflows when creating a pull request which reflect the type of development model used in the project you are contributing to;
+1. Pull request from a branch within a repository and,
+2. Pull request from a forked repository.
+
+> ## About forks
+> Before we get into understanding pull requests, we should first get to grips with what a fork is, and how it differs from a branch.
+> - By default, a public repository can be seen by anyone but only the owner can make changes e.g. create new commits or branches.
+> - `Forking` a repository means creating a copy of it in your own GitHub account.
+> - This copy is fully under your control, and you can create branches, push new commits, etc., as you would do with any other of your repos.
+> - `fork` is a GitHub concept and not Git.
+> - Forks are related to the original repository, and the number of forks a given repository has can be seen in the upper right corner of the repo page.
+> - If you have some changes in your fork that you want to contribute to the original repo, you open a `pull request`.
+> - You can bring changes from an upstream repository to your local fork.
+>
+>Essentially, the way you use pull requests will depend on what permissions you have for the repository you are contributing to. If the repository owner has not granted you write permission, then you will not be able to create and push a branch to that repository. Conversely, anyone can fork an existing repository and push changes to their personal repository.
```

Comment:
Is this legacy text? Does not seem to fit in here - certainly not in the callout.

Reply:
I agree it should probably not be in the callout. I do think it is relevant to the 2 different PR workflows though. I have moved it above the callout.